### DISCUSSION TITLE
期待大家一起贡献Serverless Devs对应的模型内容，一起推动Serverless的开发规范前进

### DISCUSSION THREAD

**Discussion by anycodes** [Serverless-Devs/Serverless-Devs#259] [Show and tell]
模型目前已经完成了第一版本的编写：https://github.com/Serverless-Devs/Serverless-Devs/tree/docs/spec

当然，在这个版本中还有很多不完善的点，包括不限于：
1. 模型中可能存在一定的语法问题、表达错误，别字的情况；
2. 模型中的图片可能存在过于粗糙的情况；
3. 模型文件缺少对应的英文表达，即英文文档；
4. .......

所以，欢迎各位小伙伴参与贡献该spec，一起推动Serverless的开发规范前进

贡献的过程可以参考：

> 有关详细信息，请参阅[贡献指南](../CONTRIBUTING.md)。

针对 spec 的贡献也可以参考以下内容：
- 将 Serverless Devs 仓库 fork 到自己的账号/组织下；
- 对 spec 内容进行修改，更新，完善；
- 对对应版本下的`readme.md`进行更新，添加自己到`作者`->`贡献者`中；
- 提`Pull requests`到仓库`Serverless-Devs/Serverless-Devs`的`docs`分支下；并添加 [Anycodes](https://github.com/anycodes) 、 [hanxie](https://github.com/hanxie-crypto) 等作为Reviewers，同时在Comment中填写好更新理由；

所有参与贡献的小伙伴，都将会成为Serverless Devs仓库的贡献者，并获得以下好处：
- 参与了Serverless Devs开发者社区，将会获得在Serverless Devs的社区中演讲，布道的机会；
- 将会不定期通过Serverless Devs的官方对外公告参与Serverless Devs社区相关贡献的名单，提升个人影响力；
- 将会获得由Serverless Devs社区提供的精美小礼品（例如签名书籍，Serverless口罩，Serverless文化衫.......）

**Comment by lowkeyrd** [comment]
先提几个concern的点，后面再继续补充
1. xxx  model这个字眼不太好，这个主要还是提serverless devs的使用/开发规范。xxx model显得太高大上了，让人以为是要提行业标准，但结果用户一看就是一套schema，会让人觉得有吹牛成分。
2. registry我觉得不要单独列出来，我们是要给用户和开发讲个故事，区分不同角色就可以了：
    1）如何使用：这里说 s.yaml 这套规范
    2)  如何贡献：说明如何开发组件、应用，再如何push到registry上
3. 文字描述我感觉有点学术范了，因为我们还没到推出serverless应用标准的阶段，我看完的直观感觉是：”这玩意太复杂了吧，咋这么概念，这么多模型呢，有点想放弃了"。

serverless devs现在这套schema还是比较通用的，不过现在其实还是一套面向多云的开发者工具(只有provider这个上层抽象，所以只能说是多云)，所有能力都是组件代码来提供， serverless心智透出的太少了(我写个ecs的组件在这上面也能跑，但跟serverless没啥关系)。
我觉得什么时候我们可以提serverless应用模型了呢，如果能将build、local/remote debug、invoke、metric、logs、infrastructure、ci/cd这些过程的接口都定义了，别人基于这套接口容易进行扩展，那这个模型就成了。

以前我也非常buy-in model这东西，不过这东西会极大限制功能体验，其实是个陷阱，我们一定要谨慎
